python/usb_console/radio_session.py

```diff
@@ -59,27 +72,78 @@ def check_for_downlink(self):
         Read device output for debug messages and state variable updates. Record debug messages
         to the logging file, and update the console's record of the state.
         '''
+        
+        #connect to PAN email account  
+        mail = imaplib.IMAP4_SSL("imap.gmail.com", 993)
+        mail.login(self.username, self.password)
+        mail.select('Inbox')
 
         while self.running_logger:
-            # TODO implement
-            pass
+            #.search() searches from mail. Data gives id's of all emails.
+            type, data = mail.search(None, '(FROM "sbdservice@sbd.iridium.com")', '(UNSEEN)')
+            mail_ids = data[0]
+            id_list = mail_ids.split()
+
+            for num in id_list:
+                #.fetch() fetches the mail for given id where 'RFC822' is an Internet 
+                # Message Access Protocol.
+                typ, data = mail.fetch(num,'(RFC822)')
+                
+                #go through each component of data and print sender and subject line
+                for response_part in data:
+                    if isinstance(response_part, tuple):
+                        # converts message from byte literal to string removing b''
+                        msg = email.message_from_string(response_part[1].decode('utf-8'))
+                        #get the file attached to the message
+                        for part in msg.walk():
+                            
+                            if part.get_content_maintype() == 'multipart':
+                                continue
+
+                            if part.get('Content-Disposition') is None:
+                                continue
+                            
+                            #check if there is an attachment
+                            if part.get_filename() is not None:
+                                #get json data from email attachment
+                                attachmentContents=part.get_payload(decode=True).decode('utf8')
+                                data=json.loads(attachmentContents)
```

Comment:
It would be nice if the processing of the attachment data into a dictionary happened in a separate function. Right now, for testing purposes, we're assuming the data in the attachment is JSON, but in the future it'll be a really complicated encoding scheme. Maybe you could create a new function called process_downlink_packet(), and inside it call json.loads on the attachment contents? This also makes the downlink-processing part of the logic more easily unit-testable in the future.